### PR TITLE
[docs] `pattern` to preserve ignoring files prefixed with an underscore

### PR DESCRIPTION
**Pull request by sarah11918**
## Changes

Adds `pattern: '**/[^_]*.(md|mdx)'` to the example migrating from an existing collection and adds a note specifying that the Content Layer API does not automatically exclude files prefaced with an underscore. This could be expected behaviour since existing collections and file-based routing work this way.

## Testing

docs!

## Docs

Will update the config reference docs. I don't think this requires a patch update since it's not a correction, just an enhancement. Docs will have this extra information immediately.

### CODE REVIEW

File: packages/astro/src/@types/astro.ts [hunk 1]
```diff
@@ -2356,7 +2360,7 @@ export interface AstroUserConfig {
 		 *     const blog = defineCollection({
 		 *       // For content layer you no longer define a `type`
 		 *      type: 'content',
-		 *      loader: glob({ pattern: "**\/*.md", base: "./src/data/blog" }),
+		 *      loader: glob({ pattern: '**/[^_]*.(md|mdx)', base: "./src/data/blog" }),
```
Comment (ascorbic) [inline packages/astro/src/@types/astro.ts:2363]:
```suggestion
		 *      loader: glob({ pattern: '**/[^_]*.md', base: "./src/data/blog" }),
```
This is simpler, as it's just `md`

File: packages/astro/src/@types/astro.ts [hunk 2]
```diff
@@ -2275,6 +2275,10 @@ export interface AstroUserConfig {
 		 * export const collections = { blog, dogs };
 		 * ```
 		 *
+	   * :::note
+		 * Loaders will not automatically [exclude files prefaced with an `_`](/en/guides/routing/#excluding-pages). Use a regular expression such as `pattern: '**/[^_]*.(md|mdx)'` in your loader to to ignore these files.
```
Comment (ascorbic) [inline packages/astro/src/@types/astro.ts:2279]:
```suggestion
		 * Loaders will not automatically [exclude files prefaced with an `_`](/en/guides/routing/#excluding-pages). Use a regular expression such as `pattern: '**/[^_]*.md'` in your loader to to ignore these files.
```

File: packages/astro/src/@types/astro.ts [hunk 3]
```diff
@@ -2275,6 +2275,10 @@ export interface AstroUserConfig {
 		 * export const collections = { blog, dogs };
 		 * ```
 		 *
+	   * :::note
+		 * Loaders will not automatically [exclude files prefaced with an `_`](/en/guides/routing/#excluding-pages). Use a regular expression such as `pattern: '**/[^_]*.md'` in your loader to to ignore these files.
+		 * :::
+		 *
```
Comment (sarah11918) [inline packages/astro/src/@types/astro.ts:2281]:
```suggestion
		 *	
```
I'm not sure if this line full of hidden tabs was causing a problem, or where it came from, but I guess can't hurt to remove. :sweat_smile: 

image

File: packages/astro/src/@types/astro.ts [hunk 4]
```diff
@@ -2275,6 +2275,10 @@ export interface AstroUserConfig {
 		 * export const collections = { blog, dogs };
 		 * ```
 		 *
+	   * :::note
+		 * Loaders will not automatically [exclude files prefaced with an `_`](/en/guides/routing/#excluding-pages). Use a regular expression such as `pattern: '**/[^_]*.md'` in your loader to to ignore these files.
```
Comment (ematipico) [inline packages/astro/src/@types/astro.ts:2279]:
```suggestion
		 * Loaders will not automatically [exclude files prefaced with an `_`](/en/guides/routing/#excluding-pages). Use a regular expression such as `pattern: '**/[^_]*.md' in your loader to to ignore these files.
```

**Review by ematipico — APPROVED** [review]

**Pull request merged**